com.unity.shadergraph/Editor/Drawing/Inspector/PropertyDrawers/ShaderInputPropertyDrawer.cs

```diff
@@ -1450,10 +1457,203 @@ string GetSanitizedReferenceName(string name)
+            m_DropdownReorderableList.drawElementCallback = (Rect rect, int index, bool isActive, bool isFocused) =>
+            {
+                DropdownEntry entry = ((DropdownEntry)m_DropdownReorderableList.list[index]);
+                EditorGUI.BeginChangeCheck();
+
+                Rect displayRect = new Rect(rect.x, rect.y, rect.width / 2, EditorGUIUtility.singleLineHeight);
+                var displayName = EditorGUI.DelayedTextField(displayRect, entry.displayName, EditorStyles.label);
+
+                if (EditorGUI.EndChangeCheck())
+                {
+                    displayName = GetSanitizedDisplayName(displayName);
+                    var duplicateIndex = FindDuplicateDropdownDisplayNameIndex(entry.id, displayName);
+                    if (duplicateIndex != -1)
+                    {
+                        var duplicateEntry = ((DropdownEntry)m_DropdownReorderableList.list[duplicateIndex]);
+                        Debug.LogWarning($"Display name '{displayName}' will create the same display name as entry {duplicateIndex + 1}.");
+                    }
+                    else if (string.IsNullOrWhiteSpace(displayName))
+                        Debug.LogWarning("Invalid display name. Display names cannot be empty or all whitespace.");
+                    else if (int.TryParse(displayName, out int intVal) || float.TryParse(displayName, out float floatVal))
+                        Debug.LogWarning("Invalid display name. Display names cannot be valid integer or floating point numbers.");
+                    else
+                        dropdown.entries[index] = new DropdownEntry(GetFirstUnusedDropdownID(), displayName);
+
+                    this._postChangeValueCallback(true);
+                }
+            };
+
+            // Element height
+            m_DropdownReorderableList.elementHeightCallback = (int indexer) =>
+            {
+                return m_DropdownReorderableList.elementHeight;
+            };
+
+            // Can add
+            m_DropdownReorderableList.onCanAddCallback = (ReorderableList list) =>
+            {
+                return true;
+            };
+
+            // Can remove
+            m_DropdownReorderableList.onCanRemoveCallback = (ReorderableList list) =>
+            {
+                return list.count > DropdownNode.k_MinEnumEntries;
+            };
+
+            // Add callback delegates
+            m_DropdownReorderableList.onSelectCallback += DropdownSelectEntry;
+            m_DropdownReorderableList.onAddCallback += DropdownAddEntry;
+            m_DropdownReorderableList.onRemoveCallback += DropdownRemoveEntry;
+            m_DropdownReorderableList.onReorderCallback += DropdownReorderEntries;
+        }
+
+        void DropdownSelectEntry(ReorderableList list)
+        {
+            m_DropdownSelectedIndex = list.index;
+        }
+
+        int GetFirstUnusedDropdownID()
+        {
+            if (!(shaderInput is ShaderDropdown dropdown))
```

Comment:
This check might be redundant given that you also check for if this input is a dropdown prior to the IMGUI code that calls this and also in the other place you call this function (DropdownAddEntry)

Reply:
True, unless someone calls it in the wrong context, since there multiple types handled in here. I would prefer to leave this matching the pattern used by keywords to be honest.

Reply:
Generally the extra checks are probably overly conservative, but it's trivial code, so I would not be concerned about performance implications.